vignettes/intro_opencage.Rmd.src

```diff
@@ -0,0 +1,188 @@
+
+```{r piped}
+library(dplyr, warn.conflicts = FALSE)
+concert_df %>% oc_forward_df(location)
+```
+
+## Reverse geocoding
+
+Reverse geocoding means that we have the coordinates, latitude and longitude, and look up the most appropriate placename for those.
+
+```{r reverse, message=FALSE}
+oc_reverse_df(latitude = 51.5034070, longitude = -0.1275920)
+```
+
+Note that all coordinates sent to the OpenCage API must adhere to the [WGS 84](https://en.wikipedia.org/wiki/World_Geodetic_System) (also known as [EPSG:4326](http://epsg.io/4326)) [coordinate reference system](https://en.wikipedia.org/wiki/Spatial_reference_system) in decimal format. 
+This is the coordinate reference system used by the [Global Positioning System](https://en.wikipedia.org/wiki/Global_Positioning_System).
+There is usually no reason to send more than six or seven digits past the decimal as that then gets down to the [precision of a centimetre](https://en.wikipedia.org/wiki/Decimal_degrees).
+
+`oc_reverse_df()` is vectorised, can work with numeric vectors and data frames, supports the `output = "all"` argument and can be used with the {magrittr} pipe, as well.
+
+OpenCage always returns at most [one result](https://opencagedata.com/api#ranking) per reverse geocoding request. 
+
+## Caching
+
+OpenCage [allows and supports caching](https://opencagedata.com/api#caching). 
+In order to send requests to the API only when necessary, {opencage} uses {[memoise](https://github.com/r-lib/memoise)} to cache results inside the active R session. 
+
+```{r memoise, message=FALSE}
+system.time(oc_reverse(latitude = 10, longitude = 10))
+
+system.time(oc_reverse(latitude = 10, longitude = 10))
+
+```
+
+To clear the cache of all results, either start a new R session or call `oc_clear_cache()`.
+
+```{r clear_cache, message=FALSE}
+oc_clear_cache()
+
+system.time(oc_reverse(latitude = 10, longitude = 10))
+
+```
+
+As you probably know, cache invalidation is one of the harder things to do in computer science. 
+Therefore {opencage} only supports invalidating the whole cache and not individual records at the moment. 
+
+The underlying data at OpenCage is [updated daily](https://opencagedata.com/faq#general). 
+
+## Further information
+
+OpenCage supports a lot of parameters to either target your search area more specifically or to specify what additional information you need. 
+See the ["Customise your query"](customise_query.html) vignette for details.
+
+Besides `oc_forward_df()` and `oc_reverse_df()`, which always return a single tibble, there are also two other functions,  `oc_forward()` and `oc_reverse()`, which return other types of output. 
+Depending on what you specify as the `return` parameter, `oc_forward()` and `oc_reverse()` will return either a list of tibbles (`df_list`, the default), JSON lists (`json_list`), GeoJSON lists (`geojson_list`), or the URL with which the API would be called (`url_only`). 
+Learn more in ["Output options"](output_options.html) vignette. 
```

Comment:
this vignette could end to a reminder of where to report bugs/feature requests and where to get help (rOpenSci forum?)